docs/platforms/javascript/common/performance/instrumentation/custom-instrumentation.mdx

```diff
@@ -16,21 +16,153 @@ To get started, import the SDK.
 
 <PlatformContent includePath="enriching-events/import" />
 
-<PlatformContent includePath="performance/span-api-version" />
+There are three key functions for creating spans:
 
-## Start Span
+- [startSpan](#starting-an-active-span-startspan): Creates a new span that is active, and which is automatically ended. You'll likely want to use this function.
+- [startSpanManual](#starting-an-active-span-with-manual-end-startspanmanual): Creates a new span that is active, which has to be ended manually.
+- [startInactiveSpan](#starting-inactive-spans-startinactivespan): Creates a new span that is inactive, which has to be ended manually.
 
-By default, spans you create are considered active, which means they are put on the Sentry scope. This allows child spans and Sentry errors to be associated with that span. This is the recommended way to create spans.
+## Active vs. Inactive Spans
+
+When a new span is started, it will automatically be started as a child of the currently active span, if there is one. This means that if a span is started as an **active span**, any spans that are created inside of the callback where the span is active will be children of that span. Additionally, errors will be tied to the currently active span, if there is one.
+
+In contrast, **inactive spans** will never have children automatically associated with them. This is useful if you do not care about capturing child activity.
+
+A key constraint for active spans is that they can only be made active inside of a callback. This constraint exists because otherwise it becomes impossible to associate spans with the correct parent span when working with asynchronous code.
+
+In places where you are not able to wrap executing code in a callback (e.g. when working with hooks or similar) you have to work with inactive spans, and can combine this with [withActiveSpan](#with-active-span) to manually associate child spans with the correct parent span.
+
+<PlatformCategorySection supported={['browser']}>
+
+## Span Hierarchy in the Browser
+
+In browser environments, it is not possible to reliably keep track of the active span across asynchronous boundaries. This means that if multiple asynchronous operations are started in parallel, it is not possible to determine which span is the parent of which child span. Imagine the following example:
```

Comment:
l/m: I'd suggest addinga TLDR-like intro sentence to this section. Right now, we start with the problem and tell folks about the consequence  4 paragraphs in.

I was thinking of something along these lines. Feel free to adjust as you see fit.

```suggestion
## Span Hierarchy in the Browser

In browser environments, spans are by default collected in a flat hierarchy where every span is the direct child of the root span. You can opt into a more fine grained hierarchy but there are trade-offs. 

The key reason for keeping a flat hierarchy is because in browsers it's impossible to reliably keep track of the active span across asynchronous boundaries. This means that if multiple asynchronous operations are started in parallel, it is not possible to determine which span is the parent of which child span. Imagine the following example:
```